code/game/machinery/doors/airlock.dm

```diff
@@ -737,125 +760,132 @@ About the new airlock wires panel:
 	else
 		try_to_activate_door(user)
 
-/obj/machinery/door/airlock/CanUseTopic(mob/user)
-	if(!issilicon(user) && !isobserver(user))
-		return STATUS_CLOSE
 
-	if(emagged)
-		to_chat(user, "<span class='warning'>Unable to interface: Internal error.</span>")
-		return STATUS_CLOSE
-	if(!canAIControl() && !isobserver(user))
-		if(canAIHack(user))
-			hack(user)
-		else
-			if(isAllPowerLoss()) //don't really like how this gets checked a second time, but not sure how else to do it.
-				to_chat(user, "<span class='warning'>Unable to interface: Connection timed out.</span>")
-			else
-				to_chat(user, "<span class='warning'>Unable to interface: Connection refused.</span>")
-		return STATUS_CLOSE
+// -------------------------------------------------------------------------------------
 
-	return ..()
 
-/obj/machinery/door/airlock/Topic(href, href_list, nowindow = 0)
-	if(..())
-		return 1
 
-	var/activate = text2num(href_list["activate"])
-	switch(href_list["command"])
-		if("idscan")
-			if(isWireCut(AIRLOCK_WIRE_IDSCAN))
-				to_chat(usr, "The IdScan wire has been cut - IdScan feature permanently disabled.")
-			else if(activate && aiDisabledIdScanner)
-				aiDisabledIdScanner = 0
-				to_chat(usr, "IdScan feature has been enabled.")
-			else if(!activate && !aiDisabledIdScanner)
-				aiDisabledIdScanner = 1
-				to_chat(usr, "IdScan feature has been disabled.")
-		if("main_power")
+/obj/machinery/door/airlock/tgui_act(action, params)
+	if(..())
+		return
+	if(!issilicon(usr) && !usr.can_admin_interact())
```

Comment:
I am not sure about having `admins` in a message youll see from an IC object but I wont make a huge debate over it